temporal-testing/src/main/java/io/temporal/internal/docker/RegisterTestNamespace.java

```diff
@@ -64,6 +90,27 @@ public static void main(String[] args) throws InterruptedException {
         System.exit(1);
       }
     }
-    System.exit(0);
+  }
+
+  private static void waitForNamespace(WorkflowServiceStubs service) throws InterruptedException {
+    // We have a namespace cache on the server side which is refreshed every 10seconds. All
+    // namespace centric APIs
+    // don't use the cache, so the best way is just to wait for it to expire. Another way is to call
+    // some
+    // workflow/activity tasks poll APIs and wait till they stop failing
+    // https://github.com/temporalio/temporal/issues/1941
+    System.out.println("Waiting for the namespace propagation...");
+    Thread.sleep(10000);
```

Comment:
I think just calling `doesNamespaceExist` in a loop with a 100ms wait in between or something is a better option than a straight wait here. It can time out after however long.

Reply:
It doesn't work. It was my original solution. It will return that it exists, because that code path doesn't use a cache.

Reply:
K, for reference this is what Node does which isn't too bad https://github.com/temporalio/sdk-node/blob/main/scripts/wait-on-temporal.js#L24

Reply:
Honestly, both ways are hacks and look pretty bad. Because we are not enabling dockerized tests yet, let's leave it like this and just wait for #1941 that actually should be addressed in upcoming weeks most likely.